### ISSUE TITLE
VI editor again

### ISSUE THREAD

**Issue by stona1** [louisdh/terminal#89]
Hi,

I have seen some time topic about introducing VIM in OpenTerm. Conclusion was that there is VIM for iOS and we do not need it in Terminal. 
I think it is still quite many fields when ability to use VI or VIM inside OpenTerm will be helpful. Starting from simple learning using VI inside Terminal going thru possibility just edit files inside OpenTerm and finishing on making thanks this OpenTerm experience closer to real terminal. What you think?

BR
Arek

**Comment by IMcD23** [comment]
I agree that vi/vim would be an awesome addition to the app. There is additional work required until we will be able to support it. Before OpenTerm can run utilities that have a more advanced interface like this one, it will need additional improvements to its handling of escape codes. See #85 and #5.

We will also need to get the editor compiling in the ios_system project.

**Comment by stona1** [comment]
Thank you for your clarification. I did not realised that it so complicated. I keep my fingers crossed that people with enough skills will add this functionality :)

**Comment by holzschu** [comment]
I'm using vim intensely on my iPad. The current setting, "share filename" (enter) "open in iVim" (click) from OpenTerm, or ":ido" (enter), pick document (click) from iVim works pretty well for me (YMMV). Both edit the files inside OpenTerm sandbox.

In addition to compiling vim (difficult) and supporting escape codes to write anywhere on screen, a practical vim inside OpenTerm would require changing the on-screen keyboard, to add at least the Escape key.
